### DISCUSSION TITLE
How should I reduce my flush operations?

### DISCUSSION THREAD

**Discussion by 12932** [juicedata/juicefs#2757] [Q&A]
I have a hobby project where I have a Postgres database, on JuiceFS, connected to Cloudflare R2,  ([followed this guide for the setup](https://github.com/centminmod/centminmod-juicefs#mount-the-juicefs-formatted-r2-s3-storage)) the database receives dozens of very tiny writes per second, spiking up to hundreds per second. I noticed that my Cloudflare R2 dashboard is showing many thousands of Class A operations (writes, puts, etc) after only running for a few hours.

My mount config is as follows:

```
-cache-dir /home/juicefs_cache \
--cache-size 102400 \
--buffer-size 1024 \
--open-cache 0 \
--attr-cache 1 \
--entry-cache 1 \
--dir-entry-cache 1 \
--cache-partial-only false \
--free-space-ratio 0.1 \
--writeback \
--backup-meta 1h \
--no-usage-report \
--max-uploads 10 \
--max-deletes 2 \
--backup-meta 1h \
--log /var/log/juicefs.log \
--get-timeout 300 \
--put-timeout 900 \
--io-retries 90 \
--prefetch 4
```

Here is a screenshot of my juicefs stats --verbosity=1

![image](https://user-images.githubusercontent.com/68835423/190170574-514591b6-c21d-42e8-86aa-703437286050.png)

I think it's constantly writing to object storage, and would like it to write less frequently, in larger chunks. Should I just increase my block size? Would increasing my cache and/or buffer size assist in this situation? My machine has 16GB of RAM, and about 64GB of SSD disk space.

**Comment by xiaogaozi** [comment]
Please confirm the following questions:

1. How is your data written to JuiceFS?
2. What is the average size of the files written?

It seems that the amount of data written to the object storage is around 10KiB-20KiB (estimated by `object.put / object.put_c`), such a small amount of writing will cause frequent requests to the object storage.

**Reply by 12932** [reply]
1) I just have the Postgres data directory on the mounted JuiceFS volume, so Postgres is constantly writing and reading to this directory
2) I'm not exactly sure how Postgres does writes to disk, I know that the default page size for Postgres is 8192 bytes

So would decreasing my JuiceFS block-size assist in this situation?

**Reply by xiaogaozi** [reply]
It is not recommended to use JuiceFS as the underlying storage for PostgreSQL. On the one hand, the performance will be poor, and on the other hand, there will be many requests for object storage due to a large number of small IO writes. Adjusting the block size of JuiceFS will not improve this scenario.

**Reply by 12932** [reply]
😭😭😭😭😭😭😭😭😭

**Reply by holygrolli** [reply]
what is the reason behind the statement "It is not recommended to use JuiceFS as the underlying storage for PostgreSQL"? My assumption was we have a reliable filesystem. What disadvantages do we face when using this for databases (like in a CSI driver scenario for any k8s persistent volume)
